main/test/flix/Test.Exp.Match.Record.flix

```diff
@@ -0,0 +1,59 @@
+/*
+ *  Copyright 2023 Jakob Schneider Villumsen
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+mod Test.Exp.Match.Record {
+
+    @test
+    def testMatchEmptyRecord01(): Bool = match { } {
+        case { } => true
+    }
+
+    @test
+    def testMatchEmptyRecord02(): Bool = match { } {
+        case { } => true
+        case _   => false
+    }
+
+    /*
+    @test
+    def testMatchEmptyRecord03(): Bool = match { x = 1 } {
+        case { x = 1 } => true
+        case { x = 2 } => false
+    }
+
+    @test
+    def testMatchEmptyRecord04(): Bool = match { x = 1 } {
+        case { x = 2 } => false
+        case { x = 1 } => true
+    }
+
+    @test
+    def testMatchEmptyRecord05(): Bool = match { x = 3 } {
+        case { x = 1 } => false
+        case { x = 2 } => false
+        case { x = 3 } => true
+    }
+
+    @test
+    def testMatchEmptyRecord06(): Bool = match { x = 3 } {
+        case { x = 1 } => false
+        case { x = 2 } => false
+        case { x = 3 } => true
+        case { x = 4 } => false
+    }
+    */
```

Comment:
Do you intend to allow:

```
case { x = foo, y = bar }
```

and what about:

```
case { x, y }
````

and what about

```
case { x = 1 | r }
```

?

Reply:
Yes, but I see those as mixtures of the individual patterns outlined in `ParsedAst.RecordPattern`. However, once I dig further in, I may realize it doesn't quite work that way.

Reply:
> What does other languages do?

I'll take a look at OCaml :)